src/GitHub.Api/Managers/Downloader.cs

```diff
@@ -67,7 +67,7 @@ class PairDownloader
             private readonly IFileSystem fs;
             private readonly CancellationToken cancellationToken;
 
-            private volatile int finishedTaskCount;
+            private int finishedTaskCount;
```

Comment:
I didn't notice this in the first pull request and I'm noticing it now. What makes these variables `volatile`? And why is that variable no longer?

Reply:
`volatile` is a hint to the compiler that this field will be touched by multiple threads, so it won't reorder operations for optimization purposes. Normally the compiler could try to optimize things by reordering operations that aren't related to each other in a method:

```
a = true;
if (b == 2) {}
```

These two operations can be run in any order and there's no guarantee that the compiler will do them in this particular order, since it doesn't matter what the order is, they aren't dependent on each other. But when you have fields that are touched from multiple places, two operations that are seemingly unrelated might suddenly have side effects that affect each other in ways that the compiler can't predict, and reordering them might introduce subtle bugs. Like for instance some other thread changing the value of `b` depending on whether `a` is true or not - then the order of setting `a` and reading `b` would be important. This isn't a real world example but it's the gist of it.

The reason I added `volatile` is because `finishedTaskCount` gets written to every time a task finishes. The reason I took it out is because  `Interlocked.Increment` already guarantees atomicity of the operation in a threaded environment and that's the only time this field is touched, so marking it as `volatile` is pointless (and the compiler helpfully points this out).